src/file/metadata.rs

```diff
@@ -354,8 +374,391 @@ impl ColumnChunkMetaData {
+        // variable-length byte arrays do not include a length prefix.
+        //
+        // Instead of using actual decoder, we manually convert values.
+        let res = match physical_type {
+          Type::BOOLEAN => {
+            let min = min.map(|data| data[0] == 1);
+            let max = max.map(|data| data[0] == 1);
+            Statistics::Boolean { order: order, min: min, max: max, nulls: null_count }
+          },
+          Type::INT32 => {
+            let min = min.map(|data| LittleEndian::read_i32(&data));
+            let max = max.map(|data| LittleEndian::read_i32(&data));
+            Statistics::Int32 { order: order, min: min, max: max, nulls: null_count }
+          },
+          Type::INT64 => {
+            let min = min.map(|data| LittleEndian::read_i64(&data));
+            let max = max.map(|data| LittleEndian::read_i64(&data));
+            Statistics::Int64 { order: order, min: min, max: max, nulls: null_count }
+          },
+          Type::INT96 => {
+            // INT96 statistics may not be correct, because comparison is signed
+            // byte-wise, not actual timestamps. It is recommended to ignore min/max
+            // statistics for INT96 columns.
+            let min = min.map(|data| {
+              assert_eq!(data.len(), 12);
+              unsafe {
+                let raw = ::std::slice::from_raw_parts(data.as_ptr() as *mut u32, 3);
+                Int96::from(Vec::from(raw))
+              }
+            });
+            let max = max.map(|data| {
+              assert_eq!(data.len(), 12);
+              unsafe {
+                let raw = ::std::slice::from_raw_parts(data.as_ptr() as *mut u32, 3);
+                Int96::from(Vec::from(raw))
+              }
+            });
+            Statistics::Int96 { order: order, min: min, max: max, nulls: null_count }
+          },
+          Type::FLOAT => {
+            let min = min.map(|data| LittleEndian::read_f32(&data));
+            let max = max.map(|data| LittleEndian::read_f32(&data));
+            Statistics::Float { order: order, min: min, max: max, nulls: null_count }
+          },
+          Type::DOUBLE => {
+            let min = min.map(|data| LittleEndian::read_f64(&data));
+            let max = max.map(|data| LittleEndian::read_f64(&data));
+            Statistics::Double { order: order, min: min, max: max, nulls: null_count }
+          },
+          Type::BYTE_ARRAY | Type::FIXED_LEN_BYTE_ARRAY => {
+            Statistics::Binary { order: order, min: min, max: max, nulls: null_count }
+          }
+        };
+
+        Some(res)
+      },
+      None => None
+    }
+  }
+
+  /// Returns column order based on physical and logical types.
+  /// This is used when min_value/max_value are set for statistics.
+  ///
+  /// Copied from `parquet.thrift`, see the file for more details on order of values.
+  fn column_order(physical_type: Type, logical_type: LogicalType) -> ColumnOrder {
```

Comment:
perhaps this can be a method for `ColumnDescriptor`?

Reply:
This might have been the wrong name for `ColumnOrder`. This is not a sort order, this is an order which should be used when comparing current value with `min` and `max` from statistics. I might nee to rename it.

Reply:
I will rewrite it, so it will be a method on `TypeDefinedOrder` struct as part of `ColumnOrder` enum. Something like this:
```rust
// Sort order for statistics, to define `min <= value <= max`.
enum SortOrder {
  SIGNED,
  UNSIGNED,
  UNKNOWN
}

// Column order for each leaf column.
enum ColumnOrder {
  TypeDefinedOrder,
  Undefined
}

impl ColumnOrder {
  pub fn sort_order(&self, field: &Type) -> SortOrder;
}
```

I just need to figure out how to handle legacy `min`/`max` values as opposed to `min_value`/`max_value`. I will follow the reference: https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

Reply:
Sounds good. Maybe also take a look at how C++ handles this: https://github.com/apache/parquet-cpp/blob/master/src/parquet/metadata.cc#L53.

I'll take a look too.

Reply:
Thanks! I had a look, and this makes clear when to use type defined order, because I was a bit confused with parquet-mr code. Will update accordingly.